authentication/LaravelInertia.js

```diff
@@ -0,0 +1,148 @@
+ *
+ * A successful login will result in a 302 redirect. If this happens, a GET request is made to the redirect URL.
+ *
+ * Every request made by this script is logged separately to the History tab.
+ */
+
+
+function authenticate(helper, paramsValues, credentials) {
+
+  var AuthenticationHelper = Java.type('org.zaproxy.zap.authentication.AuthenticationHelper');
+  var HttpRequestHeader = Java.type("org.parosproxy.paros.network.HttpRequestHeader");
+  var HttpHeader = Java.type("org.parosproxy.paros.network.HttpHeader");
+  var URI = Java.type("org.apache.commons.httpclient.URI");
+
+  var targetURL = paramsValues.get("Target URL");
+  var baseURL = targetURL.match(/^(.+?[^\/:](?=[?\/]|$))/i)[1];
+
+  //
+  // First, make a GET request to the login page to get and extract the
+  // csrfmiddlewaretoken from it.
+  //
+
+  // Build message.
+  var firstRequestURI = new URI(targetURL, false);
+  var firstRequestMethod = HttpRequestHeader.GET;
+  var firstRequestMainHeader = new HttpRequestHeader(firstRequestMethod, firstRequestURI, HttpHeader.HTTP11);
+  var firstMsg = helper.prepareMessage();
+  firstMsg.setRequestHeader(firstRequestMainHeader);
+
+
+  // Send message.
+  helper.sendAndReceive(firstMsg, false);
+
+  // Add message to ZAP history.
+  AuthenticationHelper.addAuthMessageToHistory(firstMsg);
+
+
+  // Get the csrf token from the response.
+  var csrfTokenValueRegEx = /XSRF-TOKEN=([A-Za-z0-9]*%3D)/i;
+
+  var csrfTokenValue = firstMsg.getResponseHeader().toString().match(csrfTokenValueRegEx)[1];
+
+
+  // Get the csrf token from the response.
+  var cookieSessionRegEx = /osdo_session=([A-Za-z0-9]*%3D)/i;
+  var cookieSessionValue = firstMsg.getResponseHeader().toString().match(cookieSessionRegEx)[1];
+
+
+  // Get Inertia version
+  var dataPageRegEx = /<div id=\"app\" data-page=\"([^\"]+)\"/i
+  var dataPageValue = firstMsg.getResponseBody().toString().match(dataPageRegEx)[1];
+
+  var dataPageJsonString = dataPageValue.replace(/&quot;/g, '"');
+  var dataPageObject;
+
+  dataPageObject = JSON.parse(dataPageJsonString);
+
+  if (dataPageObject) {
+    var inertiaVersion = dataPageObject.version;
+  }
+
+  // Now, make a POST request to the login page with user credentials and
+
+  var secondRequestURI = new URI(targetURL, false);
+  var secondRequestMethod = HttpRequestHeader.POST;
+  var secondRequestHeader = new HttpRequestHeader(secondRequestMethod, secondRequestURI, HttpHeader.HTTP11);
+
+  var secondMsg = helper.prepareMessage();
+  secondMsg.setRequestHeader(secondRequestHeader);
+
+  // add headers
+  secondMsg.getRequestHeader().setHeader("X-XSRF-TOKEN", decodeURIComponent(csrfTokenValue));
+  secondMsg.getRequestHeader().setHeader("Content-Type", "application/json");
+  secondMsg.getRequestHeader().setHeader("X-Requested-With", "XMLHttpRequest");
+  secondMsg.getRequestHeader().setHeader("Referer", "https://app.opensecdevops.com/login");
```

Comment:
I'd expect this to use `baseURL`, if not possible it should be a required parameter.

Reply:
Yes this a big error of the testing sorry